app/json-renderer.js

```diff
@@ -0,0 +1,21 @@
+import createJSON from './lib/fg/json'
+
+const isJSON = function (str) {
+  if (str === "" || str === ' ') return false
+  str = str.replace(/\\(?:["\\\/bfnrt]|u[0-9a-fA-F]{4})/g, '@')
+  str = str.replace(/"[^"\\\n\r]*"|true|false|null|-?\d+(?:\.\d*)?(?:[eE][+\-]?\d+)?/g, ']')
+  str = str.replace(/(?:^|:|,)(?:\s*\[)+/g, '')
+  return (/^[\],:{}\s]*$/).test(str)
```

Comment:
Would it be simpler to just do `JSON.parse(str)` and return false if it throws?

Reply:
let's say there's a very large JSON string, parsing it to the memory might even cause the browser to crash just to know that it's json, instead just use a regular expression to understand if it is of type json or not.

Reply:
That's true. In that case, I'd suggest we check the string length first, and return false if it's >2mb. It'd make sense not to handle huge JSONs anyway, right?

Reply:
yea that can be done.

Reply:
@pfrazee can you put the exact bounds for the max length of the string, i can submit a few commits.